README.md

```diff
@@ -1,3 +1,97 @@
 # RobustLeastSquares
 
-[![Build Status](https://travis-ci.org/andyferris/RobustLeastSquares.jl.svg?branch=master)](https://travis-ci.org/andyferris/RobustLeastSquares.jl)
+[![Build Status](https://travis-ci.org/FugroRoames/RobustLeastSquares.jl.svg?branch=master)](https://travis-ci.org/FugroRoames/RobustLeastSquares.jl)
+
+**RobustLeastSquares** attempts to make robust estimation using least-squares
+problems painless and simple. It provide these useful features:
+
+1. A set of linear equation solvers, to perform linear least-squares optimization steps.
+2. A set of *M*-estimators for robust estimation. An estimator is used to remove
+   outliers by reducing the cost (loss) function for abnormally large residuals, and
+   gets its name from the statistical method of estimating parameters from
+   certain statistical distributions (e.g. Gaussian statistics correspond to the
+   standard *L2*-estimator, etc).
+3. A high-level optimization routine that performs iteratively reweighted least-squares,
+   solving the effectively non-linear optimization of the cost/loss function by
+   a series of quadratic minimization steps.
+
+### Solvers
+
+The package has three in-built solvers - *QR* factorization of the ordinary
+least-squares problem, the normal-form of the least-squares problem, and conjugate
+gradient on the former.
+
+We are trying to find *x* such that *|r|* is minimized where the residual *r = Ax - b* (i.e. find the *x* so
+that *Ax* is as close as possible to *b* - if *A* is invertible, the solution is
+*inv(A).b* and *r = 0*). The equivalent "normal" form is *A'Ax - A'b*, where *A'A* is now
```

Comment:
About `A\b`, that's good Julia even if `A` is not invertible, and is a kind-of magical function that does different thing depending on `A` and `b`. Here, I was trying to specifically give people a handle of what happens in a very simple case in mathematical notation.